src/viam/sdk/components/camera/camera.hpp

```diff
@@ -83,6 +84,66 @@ class Camera : public Component {
         response_metadata metadata;
     };
 
+    /// @brief UTF-8 encoding of 'DEPTHMAP' used in the header of FORMAT_RAW_DEPTH bytes payload.
+    static const uint64_t MAGIC_NUMBER = 0x44455054484D4150ULL;
+
+    /// @brief Number of bytes of the header for FORMAT_RAW_DEPTH payloads
+    static const auto HEADER_SIZE = 8 /*magic num*/ + 8 /*width*/ + 8 /*height*/;
+
+    /// @struct depth_map
+    /// @brief Represents the dimensions and depth values of a depth map.
+    ///
+    /// depth_map holds the width and height of a depth map, along with a vector
+    /// of actual depth values. Each depth value is a 16-bit unsigned integer representing
+    /// the distance from the camera to a point in the scene.
+    struct depth_map {
+        uint64_t width;                      ///< Width of the depth map in pixels.
+        uint64_t height;                     ///< Height of the depth map in pixels.
+        std::vector<uint16_t> depth_values;  ///< A flat vector of depth values.
+
+        /// Default constructor
+        depth_map() = default;
+
+        /// Parameterized constructor for initializing depth_map with dimensions and depth values.
+        /// @param w Width of the depth map.
+        /// @param h Height of the depth map.
+        /// @param values A vector of depth values corresponding to each pixel in the depth map.
+        depth_map(uint64_t w, uint64_t h, std::vector<uint16_t> values)
+            : width(w), height(h), depth_values(std::move(values)) {}
+    };
+
+    ///
+    /// Encodes the dimensions and depth values of a depth map into a raw binary format
+    /// (MIME type FORMAT_RAW_DEPTH).
+    ///
+    /// This function takes the width and height of a depth map, along with a vector
+    /// of depth values, and encodes this information into a binary blob. The binary
+    /// format consists of "magic number" header (UTF-8 encoding for 'DEPTHMAP' in big-endian),
+    /// then the width and height encoded as 64-bit unsigned integers, followed by the depth
+    /// values encoded as 16-bit unsigned integers (big-endian format). This format is suitable
+    /// for serialization and transmission of depth map data through gRPC.
+    ///
+    /// @param depth_map the dimensions and depth values of a depth map.
+    /// @return A std::vector<unsigned char> representing the encoded binary data of the depth
+    /// map.
+    ///         The vector includes 8 bytes for width, 8 bytes for height, followed by 2 bytes
+    ///         per depth value.
+    /// @throws Exception: if the depth data values do not correspond to height and width.
+    ///
+    static std::vector<unsigned char> encode_depth_map(const Camera::depth_map& m);
```

Comment:
Hmm... I think probably this should be a non-static method on `depth_map`. While I like the shorter path of the method being on `Camera`, it seems a bit strange to have it be a static method that only takes a single `depth_map` object as an argument. And if `depth_map` ever were used by other resources such that we wanted to decouple it from `camera`, it would make sense to move the `encode`/`decode` methods along with it.

If we do end up going that route, it would make sense for `decode_depth_map` to similarly move onto `depth_map` (though of course that must remain static!).

Reply:
Also interested to know if this needs to be part of the public API? It feels like an implementation detail. That might argue for keeping it a free function rather than part of `depth_map` (if that class even continues to exist, per my xtensor comment).

Reply:
I take back my comment about it being part of the public API. My comment was based on my assumption that `Camera` methods were actually going to be trading in `depth_map`s, in which case I wouldn't have expected this part of the transformation pair to need to be visible to users. However, per my newer understanding about what is being done here, I agree that it needs to be public. I still have some questions though about how users are expected to know what is and isn't a buffer containing a depth map that they should be passing to `Camera::decode_depth_map`, or what they are expected to do with a buffer containing an encoded depth map after they call `Camera::encode_depth_map`.